### DISCUSSION TITLE
Etichette con valori null

### DISCUSSION THREAD

**Discussion by SabrinaCasucci** [opendatasicilia/HfcQGIS-md#70] [Domande] [answered]
Ciao Totò volevo etichettare concatenando i valori di due campi (Note e Quote_fald) e quindi ho usato l’espressione:
"Note" ||  '_'  ||  "Quote_fald".

Non sempre però i valori di entrambi i campi sono compilati e quindi laddove ho valori null non viene messa l'etichetta e non ricordo più come si fa.
Grazie mille
Sabrina

**Comment by pigreco** [comment, marked as answer]
Ciao Sabrina,
è possibile ovviare al problema in diversi modi:

1. usando la funziona [concat](https://hfcqgis.opendatasicilia.it/gr_funzioni/stringhe_di_testo/stringhe_di_testo_unico/#concat) in questo modo: `concat("Note",'_', "Quote_fald")`;
2. usando l'espressione [coalesce ](https://hfcqgis.opendatasicilia.it/gr_funzioni/condizioni/condizioni_unico/#coalesce)in questo modo: `coalesce("Note",'-')||'_'|| coalesce( "Quote_fald",'-')`
in quest'ultimo caso, puoi scegliere cosa scrivere quando il valore del campo è `NULL`; io ho messo il trattino `-`, ma potresti lasciarlo vuoto mettendo `''` e quindi l'espressione diventerebbe `coalesce("Note",'')||'_'|| coalesce( "Quote_fald",'')`

![image](https://user-images.githubusercontent.com/7631137/168105792-1f0d85e0-9e58-4b17-9f41-415ca5b9d27c.png)

**Comment by SabrinaCasucci** [comment]
grazie mille!!
